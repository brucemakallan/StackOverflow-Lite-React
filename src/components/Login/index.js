import React from 'react';
import PropTypes from 'prop-types';
import '../Signup/signup.scss';

const Login = ({ onChange, onSubmit }) => {
	const createFormElement = (type, id, placeholder) => (
		<div className="form-group" key={id}>
			<input
				type={type}
				id={id}
				name={id}
				className="form-control"
				placeholder={placeholder}
				required
				onChange={onChange} />
		</div>
	);

	const formElements = [
		{
			type: 'email',
			id: 'email',
			placeholder: 'email'
		},
		{
			type: 'password',
			id: 'password',
			placeholder: 'Password'
		}
	];

	return (
		<div className="container signup-box">
			<div className="signup-image">
				<img src="https://iviidev.info/downloads/images/questions.jpg" alt="poster" />
			</div>
			<form onSubmit={onSubmit} className="signup-form mt-4">
				<h3>Login</h3>
				{formElements.map(el => createFormElement(el.type, el.id, el.placeholder))}
				<button type="submit" className="btn btn-primary">Login</button>
				<div className="mt-5 small">
					<span className="mr-2">Don't have an account?</span>
					<a href="/signup">Signup</a>
				</div>
			</form>
		</div>
	);
};

Login.propTypes = {
	onChange: PropTypes.func.isRequired,
	onSubmit: PropTypes.func.isRequired
};

export default Login;
