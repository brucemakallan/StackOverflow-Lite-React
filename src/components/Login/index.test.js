import React from 'react';
import { mount } from 'enzyme';
import Login from '.';

describe('<Login /> component', () => {
	test('renders correctly', () => {
		const wrapper = mount(<Login onChange={jest.fn()} onSubmit={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});
});
