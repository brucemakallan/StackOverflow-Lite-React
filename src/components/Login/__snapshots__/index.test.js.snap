// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component renders correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Login
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onChange": [MockFunction],
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="container signup-box"
      >
        <div
          class="signup-image"
        >
          <img
            alt="poster"
            src="https://iviidev.info/downloads/images/questions.jpg"
          />
        </div>
        <form
          class="signup-form mt-4"
        >
          <h3>
            Login
          </h3>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              id="email"
              name="email"
              placeholder="email"
              required=""
              type="email"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Login
          </button>
          <div
            class="mt-5 small"
          >
            <span
              class="mr-2"
            >
              Don't have an account?
            </span>
            <a
              href="/signup"
            >
              Signup
            </a>
          </div>
        </form>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="signup-image"
          >
            <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />
          </div>,
          <form
            className="signup-form mt-4"
            onSubmit={[MockFunction]}
          >
            <h3>
              Login
            </h3>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[MockFunction]}
                placeholder="email"
                required={true}
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[MockFunction]}
                placeholder="Password"
                required={true}
                type="password"
              />
            </div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Login
            </button>
            <div
              className="mt-5 small"
            >
              <span
                className="mr-2"
              >
                Don't have an account?
              </span>
              <a
                href="/signup"
              >
                Signup
              </a>
            </div>
          </form>,
        ],
        "className": "container signup-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="signup-image"
          >
            <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />,
            "className": "signup-image",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "poster",
                "src": "https://iviidev.info/downloads/images/questions.jpg",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <form
            class="signup-form mt-4"
          >
            <h3>
              Login
            </h3>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="email"
                required=""
                type="email"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="password"
                name="password"
                placeholder="Password"
                required=""
                type="password"
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Login
            </button>
            <div
              class="mt-5 small"
            >
              <span
                class="mr-2"
              >
                Don't have an account?
              </span>
              <a
                href="/signup"
              >
                Signup
              </a>
            </div>
          </form>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Login
              </h3>,
              Array [
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[MockFunction]}
                    placeholder="email"
                    required={true}
                    type="email"
                  />
                </div>,
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="password"
                    name="password"
                    onChange={[MockFunction]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />
                </div>,
              ],
              <button
                className="btn btn-primary"
                type="submit"
              >
                Login
              </button>,
              <div
                className="mt-5 small"
              >
                <span
                  className="mr-2"
                >
                  Don't have an account?
                </span>
                <a
                  href="/signup"
                >
                  Signup
                </a>
              </div>,
            ],
            "className": "signup-form mt-4",
            "onSubmit": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <h3>
                Login
              </h3>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
              },
              "ref": null,
              "rendered": Array [
                "Login",
              ],
              "type": "h3",
            },
            Object {
              "instance": <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="email"
                  name="email"
                  placeholder="email"
                  required=""
                  type="email"
                />
              </div>,
              "key": "email",
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[MockFunction]}
                  placeholder="email"
                  required={true}
                  type="email"
                />,
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <input
                    class="form-control"
                    id="email"
                    name="email"
                    placeholder="email"
                    required=""
                    type="email"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "id": "email",
                    "name": "email",
                    "onChange": [MockFunction],
                    "placeholder": "email",
                    "required": true,
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                />
              </div>,
              "key": "password",
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="form-control"
                  id="password"
                  name="password"
                  onChange={[MockFunction]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />,
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <input
                    class="form-control"
                    id="password"
                    name="password"
                    placeholder="Password"
                    required=""
                    type="password"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "id": "password",
                    "name": "password",
                    "onChange": [MockFunction],
                    "placeholder": "Password",
                    "required": true,
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <button
                class="btn btn-primary"
                type="submit"
              >
                Login
              </button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": Array [
                "Login",
              ],
              "type": "button",
            },
            Object {
              "instance": <div
                class="mt-5 small"
              >
                <span
                  class="mr-2"
                >
                  Don't have an account?
                </span>
                <a
                  href="/signup"
                >
                  Signup
                </a>
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="mr-2"
                  >
                    Don't have an account?
                  </span>,
                  <a
                    href="/signup"
                  >
                    Signup
                  </a>,
                ],
                "className": "mt-5 small",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <span
                    class="mr-2"
                  >
                    Don't have an account?
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Don't have an account?",
                    "className": "mr-2",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Don't have an account?",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": <a
                    href="/signup"
                  >
                    Signup
                  </a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Signup",
                    "href": "/signup",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Signup",
                  ],
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onChange": [MockFunction],
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="container signup-box"
        >
          <div
            class="signup-image"
          >
            <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />
          </div>
          <form
            class="signup-form mt-4"
          >
            <h3>
              Login
            </h3>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="email"
                required=""
                type="email"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="password"
                name="password"
                placeholder="Password"
                required=""
                type="password"
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Login
            </button>
            <div
              class="mt-5 small"
            >
              <span
                class="mr-2"
              >
                Don't have an account?
              </span>
              <a
                href="/signup"
              >
                Signup
              </a>
            </div>
          </form>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="signup-image"
            >
              <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />
            </div>,
            <form
              className="signup-form mt-4"
              onSubmit={[MockFunction]}
            >
              <h3>
                Login
              </h3>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[MockFunction]}
                  placeholder="email"
                  required={true}
                  type="email"
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="password"
                  name="password"
                  onChange={[MockFunction]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary"
                type="submit"
              >
                Login
              </button>
              <div
                className="mt-5 small"
              >
                <span
                  className="mr-2"
                >
                  Don't have an account?
                </span>
                <a
                  href="/signup"
                >
                  Signup
                </a>
              </div>
            </form>,
          ],
          "className": "container signup-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="signup-image"
            >
              <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />,
              "className": "signup-image",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <img
                  alt="poster"
                  src="https://iviidev.info/downloads/images/questions.jpg"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "poster",
                  "src": "https://iviidev.info/downloads/images/questions.jpg",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <form
              class="signup-form mt-4"
            >
              <h3>
                Login
              </h3>
              <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="email"
                  name="email"
                  placeholder="email"
                  required=""
                  type="email"
                />
              </div>
              <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                />
              </div>
              <button
                class="btn btn-primary"
                type="submit"
              >
                Login
              </button>
              <div
                class="mt-5 small"
              >
                <span
                  class="mr-2"
                >
                  Don't have an account?
                </span>
                <a
                  href="/signup"
                >
                  Signup
                </a>
              </div>
            </form>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Login
                </h3>,
                Array [
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      id="email"
                      name="email"
                      onChange={[MockFunction]}
                      placeholder="email"
                      required={true}
                      type="email"
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      id="password"
                      name="password"
                      onChange={[MockFunction]}
                      placeholder="Password"
                      required={true}
                      type="password"
                    />
                  </div>,
                ],
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Login
                </button>,
                <div
                  className="mt-5 small"
                >
                  <span
                    className="mr-2"
                  >
                    Don't have an account?
                  </span>
                  <a
                    href="/signup"
                  >
                    Signup
                  </a>
                </div>,
              ],
              "className": "signup-form mt-4",
              "onSubmit": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h3>
                  Login
                </h3>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                },
                "ref": null,
                "rendered": Array [
                  "Login",
                ],
                "type": "h3",
              },
              Object {
                "instance": <div
                  class="form-group"
                >
                  <input
                    class="form-control"
                    id="email"
                    name="email"
                    placeholder="email"
                    required=""
                    type="email"
                  />
                </div>,
                "key": "email",
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[MockFunction]}
                    placeholder="email"
                    required={true}
                    type="email"
                  />,
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <input
                      class="form-control"
                      id="email"
                      name="email"
                      placeholder="email"
                      required=""
                      type="email"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "email",
                      "name": "email",
                      "onChange": [MockFunction],
                      "placeholder": "email",
                      "required": true,
                      "type": "email",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <div
                  class="form-group"
                >
                  <input
                    class="form-control"
                    id="password"
                    name="password"
                    placeholder="Password"
                    required=""
                    type="password"
                  />
                </div>,
                "key": "password",
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="form-control"
                    id="password"
                    name="password"
                    onChange={[MockFunction]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />,
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <input
                      class="form-control"
                      id="password"
                      name="password"
                      placeholder="Password"
                      required=""
                      type="password"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "password",
                      "name": "password",
                      "onChange": [MockFunction],
                      "placeholder": "Password",
                      "required": true,
                      "type": "password",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <button
                  class="btn btn-primary"
                  type="submit"
                >
                  Login
                </button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "btn btn-primary",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Array [
                  "Login",
                ],
                "type": "button",
              },
              Object {
                "instance": <div
                  class="mt-5 small"
                >
                  <span
                    class="mr-2"
                  >
                    Don't have an account?
                  </span>
                  <a
                    href="/signup"
                  >
                    Signup
                  </a>
                </div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="mr-2"
                    >
                      Don't have an account?
                    </span>,
                    <a
                      href="/signup"
                    >
                      Signup
                    </a>,
                  ],
                  "className": "mt-5 small",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <span
                      class="mr-2"
                    >
                      Don't have an account?
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Don't have an account?",
                      "className": "mr-2",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Don't have an account?",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": <a
                      href="/signup"
                    >
                      Signup
                    </a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Signup",
                      "href": "/signup",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Signup",
                    ],
                    "type": "a",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
